src/main/scala/datalog/execution/ir/IRTreeGenerator.scala

```diff
@@ -109,12 +109,23 @@ class IRTreeGenerator(using val ctx: InterpreterContext)(using JITOptions) {
   def generateNaive(ast: ASTNode): IROp[Any] = {
```

Comment:
I think you should add a method `generateStratified` that calls into the existing `generateNaive` method so you can minimize the diff and also in the case we don't really want to stratify we don't lose functionality.